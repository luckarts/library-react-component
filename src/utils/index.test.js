import React from 'react';
import { shallow } from 'enzyme';
import { useWindowDimensions } from './useWindowDimensions';

let wrapper;

function fireResize(width) {
	window.innerWidth = width;
	window.dispatchEvent(new Event('resize'));
}

function EffecfulComponent() {
	const { width } = useWindowDimensions();

	return <span>{width}</span>;
}

describe('should show updated window width', () => {
	beforeAll(() => {
		wrapper = shallow(<EffecfulComponent />);
	});
	it('passes them to the img element', () => {
		expect(wrapper.text()).toEqual('1024');
	});
});

describe('should show updated window widt ', () => {
	beforeAll(() => {
		fireResize('340');
		wrapper = shallow(<EffecfulComponent />);
	});

	it('passes them to the img element 2', () => {
		expect(wrapper.text()).toEqual('340');
	});
});
