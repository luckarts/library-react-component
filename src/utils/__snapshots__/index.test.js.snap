// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should show updated window widt  passes them to the img element 2 1`] = `
<span>
  340
</span>
`;

exports[`should show updated window width passes them to the img element 1`] = `
<span>
  1024
</span>
`;
